test/table.js

```diff
@@ -1378,6 +1378,233 @@ describe('Bigtable/Table', function() {
+          setTimeoutSpy.called;
+          assert.ifError(err);
+          assert.strictEqual(response, true);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return false after 10 min if inconsistency repeats', done => {
+        this.timeout(10 * 61 * 1000); // 10 minutes
+        table.bigtable.request = function(config, callback) {
+          if (config.method === 'generateConsistencyToken') {
+            return callback(null, [{consistency_token: 'sample-token12345'}]);
+          }
+          if (config.method === 'checkConsistency') {
+            return callback(null, {consistent: false});
+          }
+        };
+
+        table.waitForReplication(function(err, response) {
+          // check checkConsistencySpy called twice after 5seconds
+          clock.tick(10 * 60 * 1000); // 10 minutes
+          assert.strictEqual(response, false);
+          done();
+        });
+
+        clock.runAll();
+      });
+    });
+  });
+
+  describe('generateConsistencyToken', function() {
+    it('should provide proper request options', function(done) {
+      table.bigtable.request = function(config) {
+        assert.strictEqual(config.client, 'BigtableTableAdminClient');
+        assert.strictEqual(config.method, 'generateConsistencyToken');
+        assert.strictEqual(config.reqOpts.name, table.id);
+        done();
+      };
+      table.generateConsistencyToken(assert.ifError);
+    });
+
+    it('should return a consistencyToken', function(done) {
+      var response = {
+        consistencyToken: 'sample-token-123456',
+      };
+
+      table.bigtable.request = function(config, callback) {
+        callback(null, response);
+      };
+
+      table.generateConsistencyToken(function(err, apiResponse) {
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+  });
+
+  describe('checkConsistency', function() {
+    it('should provide the proper request options', function(done) {
+      var cToken = 'consistency-token-123';
+
+      table.bigtable.request = function(config) {
+        assert.strictEqual(config.client, 'BigtableTableAdminClient');
+        assert.strictEqual(config.method, 'checkConsistency');
+        assert.strictEqual(config.reqOpts.name, table.id);
+        assert.strictEqual(config.reqOpts.consistencyToken, cToken);
+        done();
+      };
+
+      table.checkConsistency(cToken);
+    });
+
+    describe('error', function() {
+      var cToken = 'consistency-token-123';
+      var error = new Error('err');
+
+      beforeEach(function() {
+        table.bigtable.request = function(config, callback) {
+          callback(error);
+        };
+      });
+
+      it('should return the error to the callback', function(done) {
+        table.checkConsistency(cToken, function(err) {
+          assert.strictEqual(err, error);
+          done();
+        });
+      });
+    });
+
+    describe('success', function() {
+      let responses = [
```

Comment:
Declare the variable here but set it's value in the beforeEach, especially since this value will get changed

Reply:
@kolodny if we set responses value in beforeEach(), .shift() on responses will be in-effective

Reply:
Does this mean that we should set `table.bigrable.request` in each test rather than doing it in the setup?

Reply:
No, that is for variable ```responses```

Reply:
Yes, `responses` is a variable used in `table.bigrable.request`.  I think that the variables and request setup should happen in the appropriate tests.

Reply:
Agreed with @sduskis, When I first reviewed there was only one test in this block, but that looks like it changed. The response value should be set at the start of each `it()` callback